aruco_detector:
  ros__parameters:
    dictionary: DICT_5X5_50
    marker_length_m: 0.05
    image_transport: raw          # use 'compressed' over Wi-Fi
    publish_tf: true
    frame_prefix: "aruco_"
    corner_refine_method: 1       # SUBPIX
    show_rejected: false
    publish_marker_array: true

  # DetectorParameters â€” balanced for 50 mm tags
    dp.adaptiveThreshWinSizeMin: 5
    dp.adaptiveThreshWinSizeMax: 29
    dp.adaptiveThreshWinSizeStep: 4
    dp.minMarkerPerimeterRate: 0.02     # allow smaller (farther) tags
    dp.maxMarkerPerimeterRate: 4.0
    dp.polygonalApproxAccuracyRate: 0.035
    dp.minCornerDistanceRate: 0.05
    dp.minMarkerDistanceRate: 0.05
    dp.minDistanceToBorder: 3

    dp.markerBorderBits: 1
    dp.perspectiveRemovePixelPerCell: 10
    dp.perspectiveRemoveIgnoredMarginPerCell: 0.15
    dp.minOtsuStdDev: 5.0
    dp.errorCorrectionRate: 0.6
